### PR TITLE
Wayland: do not create surfaces if connector initialized

### PR DESCRIPTION
**Pull request by al1img**
Signed-off-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

### CODE REVIEW

File: src/displayBackend/wayland/Connector.cpp [hunk 1]
```diff
@@ -75,7 +74,7 @@ void Connector::onInit(SurfacePtr surface, FrameBufferPtr frameBuffer)
 {
 	LOG(mLog, DEBUG) << "Init, name: " << mName;
 
-	if (mInitialized)
+	if (mSurface)
```
Comment (andr2000) [inline src/displayBackend/wayland/Connector.cpp]: Why don't you use isInitialized here?

Comment (al1img) [reply src/displayBackend/wayland/Connector.cpp]: Make sense

Comment (al1img) [reply src/displayBackend/wayland/Connector.cpp]: done

File: src/displayBackend/wayland/Connector.hpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ class Connector : public DisplayItf::Connector
 	 * Checks if the connector is initialized
 	 * @return <i>true</i> if initialized
 	 */
-	bool isInitialized() const override { return mInitialized; }
+	bool isInitialized() const override { return static_cast<bool>(mSurface); }
```
Comment (otyshchenko1) [inline src/displayBackend/wayland/Connector.hpp]: Looks ambiguous IMO

Comment (al1img) [reply src/displayBackend/wayland/Connector.hpp]: Why?

Comment (otyshchenko1) [reply src/displayBackend/wayland/Connector.hpp]: Certainly not consistent with what comment above says.

Comment (al1img) [reply src/displayBackend/wayland/Connector.hpp]: If connector is initialized means surface is created and mSurface keeps pointer on it.
mSurface is shared_ptr. Cast it to bool checks if shared_ptr keeps pointer to the object e.g.
if connector is initialized. Will mSurface != nullptr be less ambiguous in this place?

Comment (otyshchenko1) [reply src/displayBackend/wayland/Connector.hpp]: I understand these bits. From my point of view mSurface != nullptr is more cleaner, but perhaps static_cast<bool>(mSurface) is a common way on C++, I don't really know.
But my question was in comment. Text says "Checks if the connector is initialized and return true if initialized", but I don't see exactly such actions being performed in the function. I understood what was going on, after you had clarified me about it.

Comment (al1img) [reply src/displayBackend/wayland/Connector.hpp]: Good. I will merge as is.

Comment (otyshchenko1) [reply src/displayBackend/wayland/Connector.hpp]: Without clarifying comment in code?

Comment (arminn) [reply src/displayBackend/wayland/Connector.hpp]: I suppose mSurface != nullptr is appropriate way for this situation.

Comment (al1img) [reply src/displayBackend/wayland/Connector.hpp]: np. Done.

**Pull request merged**